### PR TITLE
Add support for pickling TrueType fonts

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/python-pillow/Pillow/issues/5825.

Changes proposed in this pull request:

 * TrueType fonts can now be pickled and unpickled

### CODE REVIEW

File: docs/releasenotes/9.0.0.rst [hunk 1]
```diff
@@ -72,6 +72,21 @@ Support has been added for the "title" argument in
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TrueType fonts may now be pickled and unpickled. For example:
+
+.. code-block:: python
+
+    from PIL import Image, ImageDraw, ImageFont
+
+    font = ImageFont.truetype("arial.ttf", size=30)
+    pickled_font = pickle.dumps(font1, protocol=pickle.HIGHEST_PROTOCOL)
```
Comment (radarhere) [inline docs/releasenotes/9.0.0.rst:85]:
```suggestion
    pickled_font = pickle.dumps(font, protocol=pickle.HIGHEST_PROTOCOL)
```

File: docs/releasenotes/9.0.0.rst [hunk 2]
```diff
@@ -72,6 +72,21 @@ Support has been added for the "title" argument in
...
 ``ImageShow.show(im, title="My Image")``.
 
+Added support for pickling TrueType fonts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TrueType fonts may now be pickled and unpickled. For example:
+
+.. code-block:: python
+
+    from PIL import Image, ImageDraw, ImageFont
```
Comment (radarhere) [inline docs/releasenotes/9.0.0.rst:82]:
```suggestion
    import pickle
    from PIL import ImageFont
```

File: docs/releasenotes/9.0.0.rst [hunk 3]
```diff
@@ -72,6 +72,22 @@ Support has been added for the "title" argument in
...
+
+    import pickle
+    from PIL import ImageFont
+
+    font = ImageFont.truetype("arial.ttf", size=30)
+    pickled_font = pickle.dumps(font, protocol=pickle.HIGHEST_PROTOCOL)
+
+    # Later...
+    unpickled_font = pickle.loads(pickled_font)
+
```
Comment (radarhere) [inline docs/releasenotes/9.0.0.rst:90]: Feel free to disagree - but does this really belong under 'API Additions'?

Comment (hugovk) [reply docs/releasenotes/9.0.0.rst:90]: Good question, does it fit better under another heading? Or just leave it out of the release notes?

Comment (radarhere) [reply docs/releasenotes/9.0.0.rst:90]: Oh, I don't mind leaving it in the release notes, I just thought that "Other Changes" might be the more appropriate place.

Comment (hugovk) [reply docs/releasenotes/9.0.0.rst]: Sounds good, moved!

File: Tests/test_pickle.py [hunk 4]
```diff
@@ -92,3 +95,48 @@ def test_pickle_tell():
...
+
+    # Act
+    draw1.text((10, 10), txt, font=font1)
+    draw2.text((10, 10), txt, font=font2)
+
+    # Assert
+    assert_image_equal(im1, im2)
+
+
+@pytest.mark.parametrize("protocol", list(range(0, pickle.HIGHEST_PROTOCOL + 1)))
```
Comment (radarhere) [inline Tests/test_pickle.py:116]:
```suggestion
@pytest.mark.parametrize("protocol", range(0, pickle.HIGHEST_PROTOCOL + 1))
```
Testing, I find that `list` isn't needed.

File: Tests/test_pickle.py [hunk 5]
```diff
@@ -92,3 +95,48 @@ def test_pickle_tell():
...
+
+    # Act: roundtrip
+    pickled_font = pickle.dumps(font, protocol)
+    unpickled_font = pickle.loads(pickled_font)
+
+    # Assert
+    helper_assert_pickled_font_images(font, unpickled_font)
+
+
+@pytest.mark.parametrize("protocol", list(range(0, pickle.HIGHEST_PROTOCOL + 1)))
```
Comment (radarhere) [inline Tests/test_pickle.py:129]:
```suggestion
@pytest.mark.parametrize("protocol", range(0, pickle.HIGHEST_PROTOCOL + 1))
```

**Review by radarhere — APPROVED** [review]

File: Tests/test_pickle.py (hunk 4 above)
Comment (hugovk) [reply Tests/test_pickle.py:116]: Yep, I did this originally but PyCharm's type checker complained:

image

(`set` also silences it.)

Which do you prefer?

**Pull request merged**